src/Engine/Renderer/Mesh/Mesh.hpp

```diff
@@ -169,7 +169,7 @@ class RA_ENGINE_API Mesh : public Displayable
     /// Update the picking render mode according to the object render mode
     void updatePickingRenderMode();
 
-  private:
+  protected:
```

Comment:
According to cpp guidelines C.133 (https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#c133-avoid-protected-data).
> protected data is a source of complexity and errors. protected data complicates the statement of invariants. protected data inherently violates the guidance against putting data in base classes, which usually leads to having to deal with virtual inheritance as well. 

They make difficult to ensure invariant on the objects each derived class must ensure the correct use and update of such members. Either let this private with a protected accessor (in charge of verifying eventual invariant) or stricly limit the protected data section (here, you pour everything below m_vao as protected, Is this intentional and mandatory ?)

Reply:
Well it was a mistake to make all the above protected. 
Nevertheless giving access directly or even thru a protected accessor allows to mess thing up with OpenGL, I keep this modification for myself.